### PR TITLE
start_with_wine(): Wait for Steam to be running

### LINKED ISSUE #58: Prepare releases and packaging
To bring this project further we can start tagging releases which also potentially will make it packable for distributions. With #57 we have a solid base for future changes we can build upon.

- Script adjustments
  - [x] Care about repo installations (#77)
  - [x] We should re-evaluate our self-update part since it will probably much more frequent be a plain download or repo installation instead of a git checkout. We should also consider a way to disable this check, maybe on path detection? (#82)
  - [x] Automate update branch detection (#63)
  - [x] Sane defaults for everything (#69)
    - Proton as default
    - Start as default
    - Update as default?
    - ETS2 as default
  - [x] Automatic current steam user detection. (#67)
  - [x] Remove waiting on user input in `start_with_wine()` (#83)
  - [x] I probably missed something…

- Github adjustments
  - [x] We should create a new GitHub action which will create a changelog since the last release and bundle everything up in case someone finds the release section. (#64)
  - [x] Start tagging releases (#78)
  - [x] Remove the binary from the repo and include it in the release tarball (#78)
  - [x] Update README (#80)
    - Dependencies > Optional (Add (python3-)pip and vdf)
    - Install (Replace current content with new installation instruction after publishing)
    - Overhaul everything as it's quite messy
    - Add us to it
  - [x] Investigate automatic bash/zsh autocompletion generation (#70, #75)
  - [x] provide installation through `setup.py` (#77)
  - [x] try to package for PyPi (#81)
  - [x] Care about PyPi installations (#80, #81)
  - [x] Provide an Issue template (#74)
    - Code blocks and spoiler for logs
  - [x] I probably missed something…

### PR DESCRIPTION
**Pull request by kakurasan**
This allows users not to press Enter key when Steam is ready.

Based on https://github.com/lhark/truckersmp-cli/pull/63#discussion_r423073700

Closes #58

### CODE REVIEW

File: truckersmp_cli/__init__.py [hunk 1]
```diff
@@ -492,21 +492,34 @@ def start_with_wine():
...
     env = os.environ.copy()
     env["WINEDEBUG"] = "-all"
     env["WINEARCH"] = "win64"
     env["WINEPREFIX"] = args.prefixdir
     env["WINEDLLOVERRIDES"] = "d3d11=;dxgi=" if not args.enable_d3d11 else ""
+
+    print("Waiting for Steam to be running...", end="")
+    sys.stdout.flush()
+    argv = (wine, "winedbg", "--command", "info process")
+    while True:
```
Comment (Lucki) [inline truckersmp_cli/__init__.py:504]: Do we need a loop here?
We expect them to have steam already running and a looping check will trigger immediately even when the user isn't logged in yet or steam finds an update and restarts itself (that's why we're listening for loginusers.vdf at proton).

**Comment by Lucki** [comment]
I don't have steam setup in wine but I guess you've successfully checked it's working?

**Comment by kakurasan** [comment]
Unfortunately its behavior was partially unexpected:
* `wine steam.exe` -> `truckersmp-cli`: It starts the game immediately [Expected]
* `truckersmp-cli` -> `wine steam.exe`:
    * It waits until `steam.exe` starts, then start the game [Expected]
    * *When an update is applied, the game doesn't start* [Unexpected] (Note: Replacing `steam.exe` with `steam.exe.old` forces updating/restarting itself)

> that's why we're listening for loginusers.vdf at proton

> Installations don't have to be in the standard path or even inside the prefix, how do we know the location?

I added the `--wine-steam-dir` option (Default is `C:\Program Files (x86)\Steam` in the prefix). The behavior is expected now even if there is an update for Steam client.

File: truckersmp_cli/__init__.py [hunk 2]
```diff
@@ -492,21 +492,46 @@ def start_with_wine():
...
+            old_timestamp = 0
+        while True:
+            try:
+                timestamp = os.stat(loginusers_vdf).st_mtime
+                if timestamp > old_timestamp:
+                    print("\r{}".format(" " * 70))
+                    break
+            except OSError:
+                pass
+            time.sleep(1)
```
Comment (Lucki) [inline truckersmp_cli/__init__.py:533]: We're now basically doing the same thing as above for proton with a different path. Can't we combine them into a unified function or something similar?

**Comment by Lucki** [comment]
I guess with a path for the steam installation we can also start steam with this script - completely similar to proton.

**Pull request merged**